components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -599,6 +599,55 @@ std::vector<mojom::FeedItemV2Ptr> GenerateChannelBlock(
   return result;
 }
 
+mojom::FeedItemMetadataPtr FromTopicArticle(
+    const Publishers& publishers,
+    const api::topics::TopicArticle& article) {
+  auto item = mojom::FeedItemMetadata::New();
+  auto id_it = base::ranges::find_if(publishers, [&article](const auto& p) {
+    return p.second->publisher_name == article.publisher_name;
+  });
+  if (id_it != publishers.end()) {
+    item->publisher_id = id_it->first;
+  }
+  item->publisher_name = article.publisher_name;
+  item->category_name = article.category;
+  item->description = article.description.value_or("");
+  item->title = article.title;
+  item->url = GURL(article.url);
+  item->publish_time = base::Time::Now();
+  item->image = mojom::Image::NewImageUrl(GURL(article.img.value_or("")));
+  return item;
+}
+
+// We use this for the Top News cluster, at the start of the feed, to match
+// (more or less) what Brave Search does.
+std::vector<mojom::FeedItemV2Ptr> GenerateTopTopicsBlock(
+    const Publishers& publishers,
+    const base::span<const TopicAndArticles>& topics) {
+  if (topics.empty()) {
+    return {};
+  }
+
+  std::vector<mojom::ArticleElementsPtr> items;
+  for (const auto& [topic, articles] : topics) {
+    if (articles.empty()) {
+      continue;
+    }
+
+    auto item = FromTopicArticle(publishers, articles.at(0));
+    items.push_back(mojom::ArticleElements::NewArticle(
+        mojom::Article::New(std::move(item), false)));
+    if (items.size() >
+        static_cast<size_t>(features::kBraveNewsMaxBlockCards.Get())) {
+      break;
+    }
+  }
```

Comment:
nit: don't need to call features::kBraveNewsMaxBlockCards.Get() multiple times.
```suggestion
  std::vector<mojom::ArticleElementsPtr> items;
  const auto max_block_size =
      static_cast<size_t>(features::kBraveNewsMaxBlockCards.Get());
  for (const auto& [topic, articles] : topics) {
    if (articles.empty()) {
      continue;
    }

    auto item = FromTopicArticle(publishers, articles.at(0));
    items.push_back(mojom::ArticleElements::NewArticle(
        mojom::Article::New(std::move(item), false)));
    if (items.size() > max_block_size) {
      break;
    }
  }
```